src/main/java/com/example/springbootapp/service/EmployeeService.java

```diff
@@ -27,36 +31,47 @@ public EmployeeService(EmployeeRepository employeeRepository, EmployeeMapper emp
     }
 
     public List<EmployeeDTO> getAllEmployees() {
-
         return employeeRepository.findAll().stream().map(employeeMapper::employeeToEmployeeDTO).collect(Collectors.toList());
     }
 
-    public EmployeeDTO getEmployeeById(long id){
-        Employee employee = employeeRepository.findById(id).orElseThrow(()-> new EmployeeNotFoundException("Employee with id " + id + " doesn't exist"));
+    public EmployeeDTO getEmployeeById(long id) {
+        Employee employee = employeeRepository.findById(id).orElseThrow(() -> new EmployeeNotFoundException("Employee with id " + id + " doesn't exist"));
         return employeeMapper.employeeToEmployeeDTO(employee);
     }
 
     @Transactional
     public Employee saveNewEmployee(EmployeeDTO employeeDTO) {
-        if (employeeDTO.isAtLeastOneParamNull(employeeDTO)){
+        if (employeeDTO.isAtLeastOneParamNull(employeeDTO)) {
             throw new EmployeeNotSavedException("Employee not saved because at least one field is null");
         }
+        Employee employee = employeeRepository.findByLastNameAndFirstName(employeeDTO.lastName(), employeeDTO.firstName());
+        if (employee != null) {
+            throw new EmployeeNotUniqueException("Employee already exists");
+        }
+        LOGGER.info("New {} {} employee is saved", employeeDTO.firstName(), employeeDTO.lastName());
         return employeeRepository.save(employeeMapper.employeeDTOtoEmployee(employeeDTO));
     }
 
     @Transactional
     public Employee updateEmployee(long employeeId, EmployeeDTO updatedEmployeeDTO) {
-        if (updatedEmployeeDTO.isAtLeastOneParamNull(updatedEmployeeDTO)){
+        if (updatedEmployeeDTO.isAtLeastOneParamNull(updatedEmployeeDTO)) {
             throw new EmployeeNotSavedException("Employee not saved because at least one field is null");
         }
-        Employee employee = employeeRepository.findById(employeeId).orElseThrow(()-> new EmployeeNotFoundException("Employee with id " + employeeId + " doesn't exist"));
+        Employee maybeExistsEmployee = employeeRepository.findByLastNameAndFirstName(updatedEmployeeDTO.lastName(), updatedEmployeeDTO.firstName());
```

Comment:
Please apply some formatting or refactoring to avoid such a long lines

Reply:
I moved the lines.